aspects/counter.go

```diff
@@ -1,33 +1,95 @@
 package ginmon
 
-import "github.com/gin-gonic/gin"
+import (
+	"time"
+
+	"github.com/gin-gonic/gin"
+)
 
 // CounterHandler is a Gin middleware function that increments a
 // global counter on each request.
-func CounterHandler(counter *CounterAspect) gin.HandlerFunc {
-	return func(c *gin.Context) {
-		counter.Inc()
-		c.Next()
+func CounterHandler(ca *CounterAspect) gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+		ctx.Next()
+		ca.inc <- tuple{
+			path: ctx.Request.URL.Path,
+			code: ctx.Writer.Status(),
+		}
 	}
 }
 
+type tuple struct {
```

Comment:
maybe this should be called `request` or something like that? `tuple` is maybe a bit too generic for what this is used for. (Not a blocker for me).